### PR TITLE
[5.1] Gate docblocks

### PR DESCRIPTION
**Pull request by lucasmichot**
Fix some typo, add one missing throws and fix one return

### CODE REVIEW

File: src/Illuminate/Auth/Access/Gate.php [hunk 1]
```diff
@@ -162,7 +164,7 @@ public function check($ability, $arguments = [])
      *
      * @param  string  $ability
      * @param  array  $arguments
-     * @return callable
+     * @return mixed
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/Access/Gate.php]: is it?

Comment (lucasmichot) [inline src/Illuminate/Auth/Access/Gate.php]: `return [$this->resolvePolicy($this->policies[get_class($arguments[0])]), $ability];` is an array

Comment (GrahamCampbell) [inline src/Illuminate/Auth/Access/Gate.php]: still callable thoigh?

Comment (lucasmichot) [inline src/Illuminate/Auth/Access/Gate.php]: This return type is either array, either callable - in this case it should be mixed

Comment (lukasgeiter) [inline src/Illuminate/Auth/Access/Gate.php]: @lucasmichot In this case it's a special kind of array, because it is, well... `callable` ;)

Comment (lucasmichot) [inline src/Illuminate/Auth/Access/Gate.php]: It's not.
In one case call_user_func_array takes a callable as first argument, in the other case it takes an array

Comment (lucasmichot) [inline src/Illuminate/Auth/Access/Gate.php]: docblock return notation describes the return type itself, not the expected usage

Comment (GrahamCampbell) [inline src/Illuminate/Auth/Access/Gate.php]: callable|array then?

Comment (lukasgeiter) [inline src/Illuminate/Auth/Access/Gate.php]: If you typehint `callable` you can pass an array aswell. So in my opinion this special array is `callable`.

Comment (lukasgeiter) [inline src/Illuminate/Auth/Access/Gate.php]: Think of it like you would document the return with an interface. Except in a more weird, PHP magic way...

Comment (lucasmichot) [inline src/Illuminate/Auth/Access/Gate.php]: Pear cut in two parts
https://github.com/lucasmichot/framework/commit/7d3a637828c03f29cd987413283375888cc36c2e

Comment (lukasgeiter) [inline src/Illuminate/Auth/Access/Gate.php]: As I see it, by adding `|array` (and in another part of the file `|string`) we might as well change `callable` to `Closure`. Isn't the whole purpose of `callable` to include all those three things?

Comment (GrahamCampbell) [inline src/Illuminate/Auth/Access/Gate.php]: @lucasmichot If you pass that particular array to is_callable, what happens?

Comment (GrahamCampbell) [inline src/Illuminate/Auth/Access/Gate.php]: if it's true, you must revert this change

Comment (lucasmichot) [inline src/Illuminate/Auth/Access/Gate.php]: Sure. https://github.com/lucasmichot/framework/commit/2dc0908a7807af34ea36e83e07f00e2eb0d525f4
There might a some return statements to change in Broadcasting and Queue then ?
https://github.com/laravel/framework/blob/5.1/src/Illuminate/Queue/Jobs/Job.php#L135
Also https://github.com/laravel/framework/blob/5.1/src/Illuminate/Support/Str.php#L196 maybe too
